sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -547,6 +547,10 @@ valueExpression
     | left=valueExpression comparisonOperator right=valueExpression                          #comparison
     ;
 
+builtinOpExpression
+    : primaryExpression (CONCAT_PIPE primaryExpression)*                                       #concat
```

Comment:
If we do you suggested, we need to write a rule like `primaryExpression (CONCAT_PIPE primaryExpression)+ ` to avoid left-recursive. But, IIUC this rule parses `a || b || c` into Concat(a, Concat(b, c))`. So, I fixed in the current way from the suggestion of @viirya.